contrib/jmx-metrics/README.md

```diff
@@ -0,0 +1,112 @@
+provided `otel.jmx.metrics.service.url`. After loading the Groovy script whose path is specified
+via `otel.jmx.metrics.groovy.script`, it will then run the script on the specified
+`otel.jmx.metrics.interval.seconds` and export the resulting metrics.
+
+### JMX Query Helpers
+
+- `otel.queryJmx(String objectNameStr)`
+   - This method will query the connected JMX application for the given `objectName`, which can
+   include wildcards.  The return value will be a `List<GroovyMBean>` of zero or more
+   [`GroovyMBean` objects](http://docs.groovy-lang.org/latest/html/api/groovy/jmx/GroovyMBean.html),
+   which are conveniently wrapped to make accessing attributes on the MBean simple.
+   See http://groovy-lang.org/jmx.html for more information about their usage.
+
+- `otel.queryJmx(javax.management.ObjectName objectName)`
+   - This helper has the same functionality as its other signature, but takes an `ObjectName`
+   instance if constructing raw names is undesired.
+
+### OpenTelemetry Synchronous Instrument Helpers
+
+- `otel.doubleCounter(String name, String description, String unit, Map<String, String> labels)`
+
+- `otel.longCounter(String name, String description, String unit, Map<String, String> labels)`
+
+- `otel.doubleUpDownCounter(String name, String description, String unit, Map<String, String> labels)`
+
+- `otel.longUpDownCounter(String name, String description, String unit, Map<String, String> labels)`
+
+- `otel.doubleValueRecorder(String name, String description, String unit, Map<String, String> labels)`
+
+- `otel.longValueRecorder(String name, String description, String unit, Map<String, String> labels)`
+
+These methods will return a new or previously registered instance of the applicable metric
+instruments.  Each one provides three additional signatures  where labels, unit, and description
+aren't desired upon invocation.
+
+- `otel.<meterMethod>(String name, String description, String unit)` - `labels` are empty map.
+
+- `otel.<meterMethod>(String name, String description)` - `unit` is "1" and `labels` are empty map.
+
+- `otel.<meterMethod>(String name)` - `description` is empty string, `unit` is "1" and `labels` are empty map.
+
+### Compatibility
+
+This metric extension supports Java 7+, though SASL is only supported where
+`com.sun.security.sasl.Provider` is available.
+
+### Configuration
+
+The following properties are supported via the command line or specified config properties file `(-config)`.
+Those provided as command line properties take priority of those contained in a properties file.
+
+| Property | Required | Description |
+| ------------- | -------- | ----------- |
+| `otel.jmx.metrics.service.url` | **yes** | The service URL for the JMX RMI/JMXMP endpoint (generally of the form `service:jmx:rmi:///jndi/rmi://<host>:<port>/jmxrmi` or `service:jmx:jmxmp://<host>:<port>`).|
+| `otel.jmx.metrics.groovy.script` | **yes** | The path for the desired Groovy script. |
```

Comment:
It doesn't block initial submission but I think we do need to follow up with a configuration file approach to allow using without running scripts. I remember elasticsearch dropping support for Groovy scripts because it's too hard to sandbox it. Their use case was more dangerous because scripts could be populated in requests themselves, but there is definitely a lot more surface than just a yaml file. I think there will be many users who would love to have this functionality but not ok with executing arbitrary groovy files when scraping, especially if we allow the file to be updated without restart which I think currently we don't allow but would be a nice follow up.

Also, noticed elasticsearch supports mustache files which is interesting - can we expect much utility out of arbitrary groovy compared to a templating language?
https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-scripting.html

Food for thought, supporting groovy and hopefully additionally files seems fine though.

Reply:
Also related to this is, is there any default export we can do that makes sense when no groovy script or even config file is provided? It'd be sweet to be able to get the jar, set the system property to point at service.url, and get something without messing around with anything else.

Reply:
A standardized config format - yes, this would be helpful and you've made a strong case for this as potentially a more accessible use case for several scenarios.  In general security concerns are obviously a high priority and planning some key or integrity checks will likely be a requirement for any runtime configuration or script update features.

default export - this would be handy and I think porting elements of the runtime metrics extension could be a reasonable candidate: https://github.com/open-telemetry/opentelemetry-java/tree/master/extensions/runtime_metrics/src/main/java/io/opentelemetry/extensions/metrics/runtime